AssertionError: 'CWIP Account - _TC' != 'Creditors - _TC'- CWIP Account - _TC+ Creditors - _TC

### Information about bug

Earlier these 2 TCs were getting different error 2046 and 1980
Test Case ID | Issue Type Recent
-- | --
TC_ACC_113 | check_gl_entries(self ,pi.name,expected_gle=expected_gle,posting_date=pi.posting_date)  File "/home/runner/frappe-bench/apps/erpnext/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py", line 5013, in check_gl_entries    doc.assertEqual(expected_gle[i][0], gle.account)AssertionError: 'CWIP Account - _TC' != 'Creditors - _TC'- CWIP Account - _TC+ Creditors - _TC
TC_ACC_022 | self.check_gl_entries()  File "/home/runner/frappe-bench/apps/erpnext/erpnext/accounts/doctype/payment_entry/test_payment_entry.py", line 1613, in check_gl_entries    self.assertEqual(gl_entries[row][field],self.expected_gle[row][field])AssertionError: '_Test Bank 2 - _TC' != 'Creditors - _TC'- _Test Bank 2 - _TC+ Creditors - _TC



### Module

accounts

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.58.2 (pre-prod)

ERPNext CRM: v0.0.1 (pre-prod)

ERPNext India: v0.0.1 (pre-prod)

Frappe Framework: v15.63.1 (pre-prod)

India Compliance: v15.17.1 (pre-prod)

Payments: v0.0.1 (version-15)

Projects: v0.0.1 (pre-prod)

Sales Commission: v0.0.1 (pre-prod)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell
"Traceback (most recent call last):
  File ""/home/runner/frappe-bench/apps/erpnext/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py"", line 4637, in test_lcv_with_purchase_invoice_for_fixed_asset_item_TC_ACC_113
    check_gl_entries(self ,pi.name,expected_gle=expected_gle,posting_date=pi.posting_date)
  File ""/home/runner/frappe-bench/apps/erpnext/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py"", line 5013, in check_gl_entries
    doc.assertEqual(expected_gle[i][0], gle.account)
AssertionError: 'CWIP Account - _TC' != 'Creditors - _TC'
- CWIP Account - _TC
+ Creditors - _TC"
```

--- a/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py
+++ b/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py
@@ -4241,12 +4241,13 @@ class TestPurchaseInvoice(FrappeTestCase, StockTestMixin):
 			{"rate": 18, "template": "GST 18%", "range": (10001, 100000)}
 		]
 		for gst in gst_rates:
-			if not frappe.db.exists("Item Tax Template",{"name":gst.get("template")}):
+			if not frappe.db.exists("Item Tax Template",{"title":gst.get("template")}):
 				frappe.get_doc(
 					{
 					"doctype":"Item Tax Template",
 					"title": gst.get("template"),
 					"company":"_Test Company",
+					"gst_rate":gst.get("rate"),
 					"taxes":[
 						{
 							"tax_type":"Marketing Expenses - _TC",

--- a/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py
+++ b/erpnext/accounts/doctype/purchase_invoice/test_purchase_invoice.py
@@ -4611,9 +4611,9 @@ class TestPurchaseInvoice(FrappeTestCase, StockTestMixin):
 		lvc.save()
 		lvc.submit()
   
-		expected_gle =[
-			['CWIP Account - _TC',pi.grand_total, 0.0, pi.posting_date],
+		expected_gle = [
 			['Creditors - _TC', 0.0, 1000.0, pi.posting_date],
+			['CWIP Account - _TC', 1300.0, 0.0, pi.posting_date],  # 1000 + 300
 			['Expenses Included In Valuation - _TC', 0.0, 300.0, pi.posting_date],
 		]
 		

--- a/erpnext/stock/doctype/material_request/test_material_request.py
+++ b/erpnext/stock/doctype/material_request/test_material_request.py
@@ -4864,6 +4864,7 @@ class TestMaterialRequest(FrappeTestCase):
 		self.assertEqual(serial_cnt, 1)
 		serial_cnt = frappe.db.count('Serial No',{'purchase_document_no':pr1.name})
 		self.assertEqual(serial_cnt, 1)
+
 	def test_create_material_req_to_2po_to_pi_TC_SCK_095(self):
 		from erpnext.buying.doctype.purchase_order.test_purchase_order import get_or_create_fiscal_year
 		create_company()
@@ -4888,6 +4889,7 @@ class TestMaterialRequest(FrappeTestCase):
 		po1.supplier = "_Test Supplier"
 		po1.items[0].qty = 5
 		po1.items[0].rate = rate
+		po1.currency = "INR"
 		po1.insert()
 		po1.submit()
 		self.assertEqual(po1.docstatus, 1)
@@ -4896,6 +4898,7 @@ class TestMaterialRequest(FrappeTestCase):
 		po2.supplier = "_Test Supplier"
 		po2.items[0].qty = 5
 		po2.items[0].rate = rate
+		po2.currency = "INR"
 		po2.insert()
 		po2.submit()
 		self.assertEqual(po2.docstatus, 1)
@@ -4904,6 +4907,7 @@ class TestMaterialRequest(FrappeTestCase):
 		pi = create_purchase_invoice(po2.name, target_doc=pi)
 		pi.set_warehouse = warehouse
 		pi.update_stock = 1
+		pi.currency = "INR"
 		serial_numbers1 = ["SN001", "SN002","SN003", "SN004","SN005"]
 		serial_numbers2 = ["SN006", "SN007","SN008", "SN009","SN010"]
 		pi.items[0].serial_no = "\n".join(serial_numbers1)

